index.js

```diff
@@ -82,12 +78,22 @@ function fastifyWebsocket (fastify, opts, next) {
     oldClose.call(this, cb)
   }
 
-  function wsHandle (handle, req, res) {
-    return handle.call(fastify, req[kWs], res)
+  function noHandle (conn, req) {
+    req[kWs].socket.close()
   }
 
-  function noHandle (req, res) {
-    req[kWs].socket.close()
+  // We monkeypatch the default route in order to
+  // support the websocket global handler
```

Comment:
I think this is a great solution to the global handler thing. We could consider dropping support for global handlers IMO, I am not really sure when you would want to effectively ignore the path, especially knowing that this library is for integrating with fastify, which definitely does not ignore the path for normal HTTP requests. Now that websockets would be handled all smartly using the router we could tell folks to migrate to using a route like `/*` if they really did want a global handler. I will leave it to the maintainers to make that call though.

Reply:
Does this monkeypatch change the semantics of when the global handler is fired? If I understand correctly, before this change, a global websocket handler would handle any websocket request not matched by the internal `fastify-websocket` router. Now, since this is going through the main fastify instance router, if a websockets request matched a route within the main router, fastify would run that handler and not this default one, right?

Reply:
You're right, I haven't considered this use case. I'll think of some way to handle it.
Personally, I'm all for dropping support for the global handler if the maintainers feel that would be the best way forward.

Reply:
@mcollina would you mind weighing in specifically on if you think keeping the global handler functionality this library previously exposed is worth it? By global, I mean "routeless" matching, where the same websockets handler would be invoked regardless of the initial request path. The alternative this PR enables would be "routeful" matching, where you have to register a route to get websocket connections, but you could use routes like `/*`if you wanted the same-ish behaviour. This PR tries to keep around the routeless matching, but it causes a bunch of the complexity around the `defaultRoute` exposure and whatnot that might be nice to drop. I think "routeful" matching makes a lot more sense and is a lot more explicit, and if I had to guess the global handler thing exists just cause it was easy and how the `ws` library tends to work.

Reply:
@mcollina To give more context regarding the issue I'm facing, the global handler is currently implemented by updating the `defaultRoute` function (_which I'll be changing by updating `find-my-way` as well to add a `setDefaultRoute` method and avoid the monkeypatching as suggested_), however this doesn't support the edge case where the `defaultRoute` function won't be called if the router might already have a route registered with the path the websocket connection is using.
To solve this, the solution I'm currently considering is to check in the `onRoute` hook if the route is a wildcard route with `GET` method, in which case I can do something like this:
```js
routeOptions.handler = (request, reply) => {
  if (request.raw[kWs]) {
    reply.hijack()
    let result
    if (isWildcardRoute) {
      result = handle.call(fastify, request.raw[kWs], request.raw)
    } else {
      result = wsHandler.call(fastify, request.raw[kWs], request)
    }
    if (result && typeof result.catch === 'function') {
      result.catch(err => request.raw[kWs].destroy(err))
    }
  } else {
    return handler.call(fastify, request, reply)
  }
}
```
However, this solution isn't complete because it's possible the user has not registered a wildcard route. I might try registering one in the plugin but there is no way to catch and ignore a duplicate route error as far as I can tell, and hooks are executed after all routes are registered so I can't register it after making sure it doesn't exist already.

Reply:
> however this doesn't support the edge case where the defaultRoute function won't be called if the router might already have a route registered with the path the websocket connection is using.
To solve this, the solution I'm currently considering is to check in the onRoute hook if the route is a wildcard route with GET method, in which case I can do something like this:

I don't understand this. Anyway, the solution looks ok, I'll see when there is a test.